lib/internal/streams/operators.js

```diff
@@ -150,6 +151,37 @@ async function * map(fn, options) {
   }
 }
 
+async function some(fn, options) {
```

Comment:
Note the semantics here, I tried to stay as close to the iterator helpers proposal as I could (closing the stream when a match is found).